### PR TITLE
Add spawnFlyingComponent argument to setVehicleWheelStates()

### PR DESCRIPTION
**Pull request by PlatinMTA**
This pull request is related to #2121. I could work in #2122 as well but I don't promise anything.

Now ``setVehicleWheelStates`` has a new boolean argument called ``spawnFlyingComponent`` just like ``setVehicleDoorState``. If it is set to true and a wheel is changed to ``2`` then it will make the wheel visually detach from the vehicle like in this video: https://youtu.be/1zpKGbvpbnY

```lua
bool setVehicleWheelStates ( vehicle theVehicle, int frontLeft, [ int rearLeft = -1, int frontRight = -1, int rearRight = -1, bool spawnFlyingComponent = false ])
```

I would really check twice this pull before mergin it. It is my first time pulling to MTA and I could be breaking the Coding Guidelines or doing something that is not the smartest. 

Thank you!

### CODE REVIEW

**Comment by Pirulax** [comment]
In the video I saw you had to interact with the vehicle to make it recalculate the physics. Is there a way to do that automatically?

File: Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp [hunk 1]
```diff
@@ -433,13 +441,14 @@ void CVehicleRPCs::SetVehicleFuelTankExplodable(CClientEntity* pSource, NetBitSt
 void CVehicleRPCs::SetVehicleWheelStates(CClientEntity* pSource, NetBitStreamInterface& bitStream)
 {
     unsigned char ucWheelStates[MAX_WHEELS];
-    if (bitStream.Read((char*)ucWheelStates, MAX_WHEELS))
+    bool          spawnFlyingComponent = false;
+    if (bitStream.Read((char*)ucWheelStates, MAX_WHEELS) && (!bitStream.Can(eBitStreamVersion::setVehicleWheelStates_SpawnFlyingComponent) || bitStream.ReadBit(spawnFlyingComponent)))
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp]:
```cpp
if (!bitStream.Read((char*)ucWheelStates, MAX_WHEELS))
    return;

if (bitStream.Can(eBitStreamVersion::setVehicleWheelStates_SpawnFlyingComponent))
    if (!bitStream.ReadBit(spawnFlyingComponent))
        return;
```

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 2]
```diff
@@ -6577,6 +6577,7 @@ bool CStaticFunctionDefinitions::SetVehicleWheelStates(CElement* pElement, int i
 
                 CBitStream BitStream;
                 BitStream.pBitStream->Write((const char*)&pVehicle->m_ucWheelStates[0], MAX_WHEELS);
+                BitStream.pBitStream->WriteBit(spawnFlyingComponent);
```
Comment (Pirulax) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:6659]: I think you need a `BitStream.Can` here as well.

**Review by Pirulax — COMMENTED** [review]
This PR would've been easier to review if you set a default value for `spawnFlyingComponent`.
Next time please try to keep code changes to minimal.
But thanks for the PR! I dont see any glaring issues.

**Comment by PlatinMTA** [comment]
> In the video I saw you had to interact with the vehicle to make it recalculate the physics. Is there a way to do that automatically?

I don't actually know. This has been default behaviour since forever, but it's inconsistent, sometimes it just falls down when other times it just mantains like that until you interact with the vehicle (even if you deleted all 4 wheels). When driving it does not happen tho.

I'll see what you reviewed in a moment, thanks.

EDIT:

> This PR would've been easier to review if you set a default value for `spawnFlyingComponent`.

What should I do next time? I didn't understand what are you referring to. Just in case the default value is ``false`` so it doesn't break backwards compatibility.

Also, shouldn't I edit the ``version.h`` file because I edited ``bistream.h``?

File: Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp (hunk 1 above)
Comment (PlatinMTA) [reply Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp]: Done

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp (hunk 2 above)
Comment (PlatinMTA) [reply Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:6659]: setVehicleDoorState() does not use it so I guess not. The only way that it'll read the bit is if it's on the same version and that only depends on the server.

**Comment by Proxy-99** [comment]
why this pr is not merged until now

**Comment by FileEX** [comment]
> why this pr is not merged until now

I think this PR can be closed because #3592  is doing the job

**Comment by Fernando-A-Rocha** [comment]
indeed this can be closed as #3592 has already been merged

**Pull request closed without merging**